pkg/controller/elasticsearch/client/model.go

```diff
@@ -133,6 +134,21 @@ func (s Shards) GetShardsByNode() map[string]Shards {
 	return result
 }
 
+// see https://github.com/elastic/cloud-on-k8s/issues/1796
+func (s *Shards) UnmarshalJSON(data []byte) error {
+	type Alias Shards
+	aux := (*Alias)(s)
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+	for i, shard := range *aux {
```

Comment:
Even though we have the link to the issue think I'd still appreciate a comment like:
`// Strip extra information from the nodeName field (eg. "cluster-node-2 -> 10.56.2.33 8DqGuLtrSNyMfE2EfKNDgg" becomes "cluster-node-2"`).